src/essreflectometry/types.py

```diff
@@ -0,0 +1,103 @@
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+Reference = NewType('Reference', str)
+Sample = NewType('Sample', str)
+Run = TypeVar('Run', Reference, Sample)
+
+
+class RawData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event time data from nexus file,
+    binned by `detector_number` (pixel of the detector frame)."""
+
+
+class WavelengthData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths computed for every event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class ThetaData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths and scattering angle computed for each event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class QData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelength, incidence angle and momentum transfer computed per event.
+    Binned by momentum transfer according to the QBins provider,
+    and by `detector_number` (pixel of the detector frame).
+    """
+
+
+class FootprintCorrected(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with weight corrected for the footprint of the beam on the sample for the
+    incidence angle of the event."""
+
+
+SpecularReflectionCoordTransformGraph = NewType(
+    'SpecularReflectionCoordTransformGraph', dict
+)
+
+
+class Histogrammed(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Histogram of event weights by momentum transfer and detector_number."""
```

Comment:
The naming is a bit unclear here. We have `QData` and  `NormalizedIofQ`. Where does `Histogrammed` fit in?

Reply:
Hm I see what you mean. Perhaps `IofQ` would be a better name instead of `Histogrammed`?